esphome/components/sensor/__init__.py

```diff
@@ -186,6 +196,7 @@ def sensor_schema(
     icon_: str,
     accuracy_decimals_: int,
     device_class_: Optional[str] = DEVICE_CLASS_EMPTY,
+    state_class_: Optional[str] = STATE_CLASS_MEASUREMENT,
```

Comment:
Nit: with this many parameters we should really enforce the use of kwargs.

```suggestion
    *, 
    state_class: Optional[str] = STATE_CLASS_MEASUREMENT,
```
And change the name of the `one_of` validator to `validate_state_class`

Reply:
why? the input is clear, if you want to force that, I suggest some general refactoring and not a one of occurance.

Reply:
> why? the input is clear by the constant naming.

Yes, when _reading_ code it's fine. But when writing (or reviewing) PRs, it would be hard to spot if device_class and state_class parameters were swapped.

Reply:
I agree that it should be done, but let's do it for all parameters, not just a random new one.
I'm considering this out of scope.

Reply:
Understood, and ok with me (thus also it being tagged as nit)